### PR TITLE
update test framework

### PR DESCRIPTION
**Pull request by mistafunk**
Get latest Catch2 release using the CMake FetchContent feature

### CODE REVIEW

File: src/common.cmake [hunk 1]
```diff
@@ -132,6 +132,24 @@ function(srl_add_dependency_maya TGT)
...
+
+function(srl_add_dependency_catch TGT)
+	FetchContent_Declare(
+			catch
+			GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+			GIT_TAG v2.13.8)
+
+	FetchContent_GetProperties(catch)
+	if(NOT catch_POPULATED)
+		FetchContent_Populate(catch)
```
Comment (chr11115) [inline src/common.cmake:145]: Consider adding message(STATUS "Fetching Catch2 from ${CATCH2_URL}...")

**Review by chr11115 — APPROVED** [review]
LGTM, just a minor UX improvement suggestion

File: src/common.cmake (hunk 1 above)
Comment (mistafunk) [reply src/common.cmake:145]: Good catch ;-)
Done.

**Comment by chr11115** [comment]
LGTM, can be merged

**Pull request merged**